src/components/action-bar/isItem.ts

```diff
@@ -0,0 +1,8 @@
+import { ListSeparator } from '../list/list-item.types';
+import { ActionBarItem } from './action-bar.types';
+
+export function isItem(
+    item: ActionBarItem | ListSeparator
+): item is ActionBarItem {
+    return !('separator' in item);
```

Comment:
I broke this out, to make it testable. There seem to be at least a few more functions or parts of functions that could be broken out and thus made more easily testable.